updated to use pivot_longer and pivot_wider functions (#736)
tidyverse/tidyr: vignettes/tidy-data.Rmd

```{r, message = FALSE}
library(tidyr)
library(dplyr)
library(stringr)

> preg2 <-preg %>% 

Reviewer: Missing a space here

Author: Fixed

>   pivot_longer(cols = c(treatmenta:treatmentb), names_to = "treatment", values_to = "n") %>% 

Reviewer: Can you please omit the name of the `cols` argument?

Author: I assume that you mean `pivot_longer((treatmenta:treatmentb), names_to = "treatment", values_to = "n")`, if so, then done.

billboard
```

To tidy this dataset, we first use pivot_longer to gether up all the `wk` columns. The column names give the `week` and the values are the `rank`s:

```{r}
billboard2 <- billboard %>% 

>   pivot_longer(cols = c(wk1:wk76), names_to = "week", values_to = "rank") %>% 

Reviewer: And no need for `c()` here

Author: Removed

  filter(!is.na(rank)) %>% 
  arrange(week, artist)
billboard2
```

> Here we use `filter(!is.na())` to drop any missing values from the rank column. In this data, missing values represent weeks that the song wasn't in the charts, so can be safely dropped.

Reviewer: Can you please use `values_drop_na = TRUE` instead?

Author: Sure. Done here and elsewhere.

## Multiple variables stored in one column

After pivoting columns, the key column is sometimes a combination of multiple underlying variable names. This happens in the `tb` (tuberculosis) dataset, shown below. This dataset comes from the World Health Organisation, and records the counts of confirmed tuberculosis cases by `country`, `year`, and demographic group. The demographic groups are broken down by `sex` (m, f) and `age` (0-14, 15-25, 25-34, 35-44, 45-54, 55-64, unknown).

```{r}
tb <- as_tibble(read.csv("tb.csv", stringsAsFactors = FALSE))
tb
```

> First we use pivot_longer to gather up the non-variable columns:

Reviewer: ```suggestion
First we use `pivot_longer()` to gather up the non-variable columns:
```

Author: Changed.

```{r}
tb2 <- tb %>% 
  pivot_longer(cols = c(-iso2, -year), names_to = "demo", values_to = "n") %>% 

>   filter(!is.na(n)) %>% 

Reviewer: And here

Author: Here too.

weather3
```

> This dataset is mostly tidy, but the `element` column is not a variable; it stores the names of variables. (Not shown in this example are the other meteorological variables `prcp` (precipitation) and `snow` (snowfall)). Fixing this requires the pivot_wider operation. This performs the inverse of pivot_longer by spreading the `element` and `value` columns back out into the columns:

Reviewer: ```suggestion
This dataset is mostly tidy, but the `element` column is not a variable; it stores the names of variables. (Not shown in this example are the other meteorological variables `prcp` (precipitation) and `snow` (snowfall)). Fixing this requires widening the data. This performs the inverse of `pivot_longer()` by pivoting the `element` and `value` columns back out into the columns:
```

```{r}
weather3 %>% 

>   pivot_wider(names_from = "element", values_from = value)

Reviewer: ```suggestion
  pivot_wider(names_from = element, values_from = value)
```
